### PR TITLE
Cooldown feature (backport) into 1.19.2

### PR DESCRIPTION
**Pull request by Skullians**
Hey,
I've backported the ultimine cooldown feature from the later versions back to 1.19.2...

This was originally a private port used on our public fabric server, however CurseForge won't allow us to publish our modpack with the private port due to your licensing / no permission

### CODE REVIEW

**Comment by Skullians** [comment]
Reopened from #150 (did the wrong branch beforehand)

File: common/src/main/resources/ftbultimine-common.mixins.json [hunk 1]
```diff
@@ -11,5 +11,6 @@
   "injectors": {
     "defaultRequire": 1
   },
-  "minVersion": "0.8"
+  "minVersion": "0.8",
+  "refmap": "ftb-ultimine-common-refmap.json"
```
Comment (desht) [inline common/src/main/resources/ftbultimine-common.mixins.json:15]: What is the purpose of this change?  With it present, the instance fails to start for me with a mixin-related exception. Removing it allowed the instance to start.

Comment (Skullians) [reply common/src/main/resources/ftbultimine-common.mixins.json:15]: huh... not sure, it's been working for me :/ I'll look into it. Good catch.

Not entirely sure why it's there 😆

**Comment by desht** [comment]
There are a couple of issues I can see from testing (other than the mixin refmap question):

* Cooldowns appear to work (as in, digging only digs one block while on cooldown)  but the progress bar isn't being displayed on the client while on cooldown.
* Missing translation `ftbultimine.server_settings.ultimine_cooldown`, visible in the `/ftbultimine serverconfig` screen.

**Comment by Skullians** [comment]
That's... odd
I've been using this mod and I get the progress bar :/

**Comment by desht** [comment]
Yeah, I tried it outside the dev environment as well (wouldn't be the first time Intellij left something cached), same deal. Cooldowns not displaying on the client :/

**Comment by Skullians** [comment]
Working for me :/

File: common/src/main/resources/ftbultimine-common.mixins.json (hunk 1 above)
Comment (Skullians) [reply common/src/main/resources/ftbultimine-common.mixins.json]: Resolved in https://github.com/FTBTeam/FTB-Ultimine/pull/151/commits/de13ac21eb5f8e83e4ccd94ff2d7ee6f1264a8ea

**Comment by Skullians** [comment]
Are you testing this on a singleplayer world or on a server?

**Comment by desht** [comment]
Think I've found the problem; cooldown is getting sync'd to the client when the server config changes.  So it works after I log out and in again (player login causes a sync).  But not if I initially set the cooldown time then try to do some ultimining.

Our current (1.21) codebase has this in `SyncConfigToServerPacket`:

image

Your PR doesn't send the `TimeType.COOLDOWN` packet. Adding that line (the 1.19 equivalent, network API is a little different) should fix that problem.

**Comment by Skullians** [comment]
Oh _right_, good catch!

```java
for (ServerPlayer player : server.getPlayerList().getPlayers()) {
                if (!sp.getUUID().equals(player.getUUID())) {
                    new SyncConfigFromServerPacket(config).sendTo(player);
                }
                new SyncUltimineTimePacket(FTBUltimineServerConfig.getUltimineCooldown(player), SyncUltimineTimePacket.TimeType.COOLDOWN.COOLDOWN).sendTo(player);         
            }
```

I assume this is fine?

**Comment by Skullians** [comment]
Sorry about that.

**Comment by desht** [comment]
No worries.  Looks good other than you `SyncUltimineTimePacket.TimeType.COOLDOWN.COOLDOWN` where `SyncUltimineTimePacket.TimeType.COOLDOWN` would be quite sufficient :)

**Comment by Skullians** [comment]
Wait wtf? 
What is my IDE on?
I can't even seem to access COOLDOWN.COOLDOWN in tab complete????????

**Comment by Skullians** [comment]
Thanks, good catch again.
Sorry.

**Comment by Skullians** [comment]
Okay!
Thanks so much for your time.

**Comment by desht** [comment]
Merged, thanks!  Will get a release out soon.

**Comment by Skullians** [comment]
Thanks so much!

**Comment by Skullians** [comment]
Do you happen to have any ETA on when you can get a release out?

**Comment by desht** [comment]
Today, with luck!

**Comment by Skullians** [comment]
Oh sweet!

**Comment by desht** [comment]
It's there:

https://www.curseforge.com/minecraft/mc-mods/ftb-ultimine-forge/files/5670778

**Comment by Skullians** [comment]
Thanks so much!

**Pull request merged**